api/cards.ts

```diff
@@ -1,9 +1,19 @@
 import axios from "axios";
+import { checkToken } from "./checkTokenMiddleware";
 
 export async function getCards(search: string) {
-  return await axios.get("http://localhost:4200/cards", {
-    params: {
-      search: search,
-    },
-  });
+  await checkToken();
+  const token = localStorage.getItem("access_token");
+  return await axios
+    .get("http://localhost:4200/cards", {
```

Comment:
it's better to store domain part in common/env variable 
```suggestion
    .get(`${host}/cards`, {
```